Added some fixes for the AWS setup guide (#47)
elastisys/compliantkubernetes: docs/operator-manual/aws.md

BASE_DOMAIN="example.com"
```

> !!! Note

Reviewer: TIL: Admonition are case-insensitive.

    We suggest exposing AWS credentials to Terraform via environment variables, so they are not accidently left on the file-system:

    ```shell
    export TF_VAR_AWS_ACCESS_KEY_ID="xyz" # Access key for AWS
    export TF_VAR_AWS_SECRET_ACCESS_KEY="zyx" # Secret key for AWS

>     export TF_VAR_AWS_SSH_KEY_NAME="~/.ssh/id_rsa.pub" # Name of the AWS keypair to use for the EC2 instances

Reviewer 2: This is confusing me. Should it be a file name, name of some object in AWS, or a path to a file? :thinking:

Author: Yes, this should be confusing because I totally messed this up. It should be a **name** of the key not the file. Great catch!

    ```shell
    export TF_VAR_AWS_ACCESS_KEY_ID="www" # Access key for AWS

>     export TF_VAR_AWS_SECRET_ACCESS_KEY="xxx" # Secret key for AWS

Reviewer 3: I don't know but we should perhaps refrain from using `xxx`.
```suggestion
    export TF_VAR_AWS_SECRET_ACCESS_KEY="xyz" # Secret key for AWS
```

    ```
    for CLUSTER in $SERVICE_CLUSTER $WORKLOAD_CLUSTERS; do
        ./bin/ck8s-kubespray init $CLUSTER default ~/.ssh/id_rsa.pub

        # This parts needs refactoring, in the meanwhile:
        sed -e 's@^---$@@' -i $CK8S_CONFIG_PATH/$CLUSTER-config/group_vars/all/all.yml
        sed -e 's@^---$@@' -i $CK8S_CONFIG_PATH/$CLUSTER-config/group_vars/k8s-cluster/k8s-cluster.yml
        sed -e 's@^etcd_kubeadm_enabled:.*@#etcd_kubeadm_enabled: false@' -i $CK8S_CONFIG_PATH/$CLUSTER-config/group_vars/all/all.yml
        echo 'ansible_user: ubuntu' >> $CK8S_CONFIG_PATH/$CLUSTER-config/group_vars/all/all.yml
        sed -e 's@.*[^_]cloud_provider:.*@cloud_provider: aws@' -i $CK8S_CONFIG_PATH/$CLUSTER-config/group_vars/all/all.yml
        sed -e "s@.*kube_oidc_auth:.*@kube_oidc_auth: true@" -i $CK8S_CONFIG_PATH/$CLUSTER-config/group_vars/k8s-cluster/k8s-cluster.yml
        sed -e "s@.*kube_oidc_url:.*@kube_oidc_url: https://dex.$BASE_DOMAIN@" -i $CK8S_CONFIG_PATH/$CLUSTER-config/group_vars/k8s-cluster/k8s-cluster.yml
        sed -e "s@.*kube_oidc_client_id:.*@kube_oidc_client_id: kubelogin@" -i $CK8S_CONFIG_PATH/$CLUSTER-config/group_vars/k8s-cluster/k8s-cluster.yml

>         sed -e "s@.*kube_oidc_username_claim:.*@kube_oidc_username_claim: email@" -i $CK8S_CONFIG_PATH/$CLUSTER-config/group_vars/k8s-cluster/k8s-cluster.yml

Reviewer: So boring. You just killed all the excitement with setting up CK8s on AWS. :smile:

>         ./bin/ck8s-kubespray init $CLUSTER aws /path/to/private/ssh/key

Reviewer: ```suggestion
        ./bin/ck8s-kubespray init $CLUSTER aws ~/.ssh/id_rsa.pub
```

This feels like a sane default and makes 95% of the documentation copy-paste-able. If the admin wants it differently, they are free to do so.

If you really hate my suggestion, then I would agree to starting the documentation with `SSH_PUBLIC_KEY_PATH=~/.ssh/id_rsa.pub` and using that throughout.

>         The key in `/path/to/private/ssh/key` must be the private key of the key referenced in `TF_VAR_AWS_SSH_KEY_NAME` above.

Reviewer: Same comment as above. I would prefer giving here a sane default and expect the administrator to change it if necessary.

```suggestion
        The key in `~/.ssh/id_rsa.pub` must be the private key of the key referenced in `TF_VAR_AWS_SSH_KEY_NAME` above.
```

Author: Alright, but then I'll change the line 
```
export TF_VAR_AWS_SSH_KEY_NAME="yyy"
```
to
```
export TF_VAR_AWS_SSH_KEY_NAME="~/.ssh/id_rsa.pub"
```
as well.

And the suggested change you made should **NOT** be pointing to your public key, it should point to your private key.

Reviewer: Just a sec. `TF_VAR_AWS_SSH_KEY_NAME` is the name of the SSH key *as known by AWS*. E.g., in my case `~/.ssh/id_rsa.pub` is known as `cklein2` in AWS.

Reviewer: > And the suggested change you made should **NOT** be pointing to your public key, it should point to your private key.

May I express my surprise? I pointed it to my public SSH key, since the private one is on a YubiKey and cannot be extracted by means known to me, and it worked.

    ```bash
    for CLUSTER in $SERVICE_CLUSTER $WORKLOAD_CLUSTERS; do
        ./bin/ck8s-kubespray apply $CLUSTER --flush-cache -e ansible_user=ubuntu
    done
    ```

>     This may take up to 20 minutes per cluster.

Reviewer: Hmm ... Should we have instead:

```
for CLUSTER in $SERVICE_CLUSTER $WORKLOAD_CLUSTERS; do
    ./bin/ck8s-kubespray apply $CLUSTER --flush-cache -e ansible_user=ubuntu >&2 > kubespray-$CLUSTER.log &
done
tail -f kubespray-*.log
```
?

I'll let you be the judge.

Author: I don't really like that, then if you try to `Ctrl+C` the cluster creation you might think that you've aborted the process but in reality you have it in the background. But it would be nice to parallelize this process some way.

@lentzi90 pointed me at this https://github.com/kubernetes-sigs/kubespray/blob/master/docs/mitogen.md which would otherwise/also be nice.

Reviewer: I agree.
